src/asymmetric/ecc.rs

```diff
@@ -0,0 +1,77 @@
+//! Elliptic curve cryptography (ECC) primitives.
+
+use winapi::shared::bcrypt::*;
+
+pub trait Curve {
+    fn as_curve(&self) -> NamedCurve;
+    fn key_bits(&self) -> u32;
+}
+
+pub struct NistP256;
+impl Curve for NistP256 {
+    fn as_curve(&self) -> NamedCurve {
+        NamedCurve::NistP256
+    }
+    fn key_bits(&self) -> u32 { 256 }
+}
+
+pub struct NistP384;
+impl Curve for NistP384 {
+    fn as_curve(&self) -> NamedCurve {
+        NamedCurve::NistP384
+    }
+    fn key_bits(&self) -> u32 { 384 }
+}
+
+pub struct NistP521;
+impl Curve for NistP521 {
+    fn as_curve(&self) -> NamedCurve {
+        NamedCurve::NistP521
+    }
+    fn key_bits(&self) -> u32 { 521 }
+}
+
+pub struct Curve25519;
+impl Curve for Curve25519 {
+    fn as_curve(&self) -> NamedCurve {
+        NamedCurve::Curve25519
+    }
+    fn key_bits(&self) -> u32 { 255 }
+}
+
+ // Required due to 1.37 being MSRV
+ // TODO: Replace with #[non_exhaustive] once we bump MSRV to 1.40

```

Comment:
The `non_exhaustive` is temporary, right? Once we have all named curves implementend, it doesn't make sense to have it non-exhaustive.

Reply:
Removed the `non_exhaustive` attribute for now.